pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeConsumptionRateManager.java

```diff
@@ -42,25 +43,35 @@
  */
 public class RealtimeConsumptionRateManager {
   private static final Logger LOGGER = LoggerFactory.getLogger(RealtimeConsumptionRateManager.class);
-  private static final RealtimeConsumptionRateManager INSTANCE = new RealtimeConsumptionRateManager();
   private static final int CACHE_ENTRY_EXPIRATION_TIME_IN_MINUTES = 10;
 
+  private static volatile RealtimeConsumptionRateManager INSTANCE;
+
   // stream config object is required for fetching the partition count from the stream
-  private final LoadingCache<StreamConfig, Integer> _streamConfigToTopicPartitionCountMap = buildCache();
+  private final LoadingCache<StreamConfig, Integer> _streamConfigToTopicPartitionCountMap;
   private volatile boolean _isThrottlingAllowed = false;
 
-  private RealtimeConsumptionRateManager() {
+  @VisibleForTesting
+  RealtimeConsumptionRateManager(LoadingCache<StreamConfig, Integer> streamConfigToTopicPartitionCountMap) {
+    _streamConfigToTopicPartitionCountMap = streamConfigToTopicPartitionCountMap;
   }
 
   public static RealtimeConsumptionRateManager getInstance() {
+    if (INSTANCE == null) {
```

Comment:
Can you use a holder class with a static final RealtimeConsumptionRateManager instead - class initialisation is atomic and the holder singleton pattern is both more concise and more efficient on ARM because there’s no need for the reads to be volatile.

Reply:
Performance is not an issue here because getting an instance is not in a critical execution path (it gets called every few hours or so). For readability, I thought using double checked singleton reads better, but after I changed it to a holder class, it looks more concise.